codec-stomp/src/test/java/io/netty/handler/codec/stomp/StompSubframeDecoderTest.java

```diff
@@ -155,4 +160,40 @@ public void testMultipleFramesDecoding() {
 
         assertNull(channel.readInbound());
     }
+
+    @Test
+    public void testValidateHeadersDecodingDisabled() {
+        ByteBuf invalidIncoming = Unpooled.buffer();
+        invalidIncoming.writeBytes(StompTestConstants.FRAME_WITH_INVALID_HEADER.getBytes(CharsetUtil.US_ASCII));
+        boolean bufAdded = channel.writeInbound(invalidIncoming);
+        assertTrue(bufAdded);
+
+        StompHeadersSubframe frame = channel.readInbound();
+        assertNotNull(frame);
+        assertEquals(StompCommand.SEND, frame.command());
+        assertTrue(frame.headers().contains("destination"));
+        assertTrue(frame.headers().contains("content-type"));
+        assertFalse(frame.headers().contains("current-time"));
+
+        StompContentSubframe content = channel.readInbound();
+        String s = content.content().toString(CharsetUtil.UTF_8);
+        assertEquals("some body", s);
+        content.release();
+    }
+
+    @Test
+    public void testValidateHeadersDecodingEnabled() {
+        EmbeddedChannel channel = new EmbeddedChannel(new StompSubframeDecoder(true));
```

Comment:
remove `EmbeddedChannel` here to ensure `channel.finish()` is called on the correct instance later on.